Source/SwiftLintFramework/Reporters/HTMLReporter.swift

```diff
@@ -30,144 +30,82 @@ public struct HTMLReporter: Reporter {
-            "\t\t\t\t\t\t<b>Severity</b>\n",
-            "\t\t\t\t\t</th>\n",
-            "\t\t\t\t\t<th style=\"width: 500pt;\">\n",
-            "\t\t\t\t\t\t<b>Message</b>\n",
-            "\t\t\t\t\t</th>\n",
-            "\t\t\t\t</tr>\n",
-            "\t\t\t</thead>\n",
-            "\t\t\t<tbody>\n", rows, "\t\t\t</tbody>\n",
-            "\t\t</table>\n",
-            "\t\t\n",
-            "\t\t<br/>\n",
-            "\t\t\n",
-            "\t\t<h2>Summary</h2>\n",
-            "\t\t\n",
-            "\t\t<table>\n",
-            "\t\t\t<tbody>\n",
-            "\t\t\t\t<tr>\n",
-            "\t\t\t\t\t<td>Total files with violations</td>\n",
-            "\t\t\t\t\t<td>\(fileCount)</td>\n",
-            "\t\t\t\t</tr>\n",
-            "\t\t\t\t<tr>\n",
-            "\t\t\t\t\t<td>Total warnings</td>\n",
-            "\t\t\t\t\t<td>\(warningCount)</td>\n",
-            "\t\t\t\t</tr>\n",
-            "\t\t\t\t<tr>\n",
-            "\t\t\t\t\t<td>Total errors</td>\n",
-            "\t\t\t\t\t<td>\(errorCount)</td>\n",
-            "\t\t\t\t</tr>\n",
-            "\t\t\t</tbody>\n",
-            "\t\t</table>\n",
-            "\t\t\n",
-            "\t\t<hr />\n",
-            "\t\t\n",
-            "\t\t<p>\n",
-            "\t\t\tCreated with\n",
-            "\t\t\t<a href=\"https://github.com/realm/SwiftLint\"><b>SwiftLint</b></a>\n",
-            "\t\t\t", swiftlintVersion, " on ", dateString, "\n",
-            "\t\t</p>\n",
-            "\t</body>\n",
-            "</html>"
-        ].joined()
+        let parameters = [
+            "VIOLATIONS": rows,
+            "TOTAL_VIOLATING_FILES": String(fileCount),
+            "TOTAL_WARNINGS": String(warningCount),
+            "TOTAL_ERRORS": String(errorCount),
+            "VERSION": swiftlintVersion,
+            "DATE": dateString
+        ]
+
+        return generateReport(parameters: parameters)
+    }
+
+    private static func generateReport(parameters: [String: String]) -> String {
+        var report = loadTemplate()
+        for (key, value) in parameters {
+            report = report.replacingOccurrences(of: "$$\(key)$$", with: value)
+        }
+
+        return report
+    }
+
+    private static func loadTemplate() -> String {
+        let path = templatePath()
+
+        #if os(Linux)
+            // swiftlint:disable:next force_try
+            let data = try! Data(contentsOf: URL(fileURLWithPath: path))
+            return String(data: data, encoding: .utf8)!
+        #else
+            // swiftlint:disable:next force_try
+            return try! String(contentsOfFile: path)
+        #endif
+    }
+
+    private static func templatePath() -> String {
+        #if SWIFT_PACKAGE
+            let components = Array(NSString(string: #file).pathComponents.dropLast() + ["Templates", "template.html"])
+            let path = components.joined(separator: "/")
+        #else
+            // swiftlint:disable:next nesting
```

Comment:
No luck in combining the paths, as it returns `nil` when running with SPM, probably because it doesn't use dynamic frameworks AFAIK.

Do you still think it's worth to get the bundle from identifier when not using SPM?